File: backend/mobile-leetnote-backend/src/main/java/com/example/leetnote_backend/model/DTO/UserProfileDTO.java
package com.example.leetnote_backend.model.DTO;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class UserProfileDTO {
    private Long userId;
    private String email;
    private String username;
    private String profileUrl;
}

File: frontend/mobile-leetnode-frontend/app/src/main/java/com/example/leetnote/data/model/UserProfileDTO.kt
package com.example.leetnote.data.model

data class UserProfileDTO(
    val id: Long,
    val email: String,
    val username: String,
    val profileUrl: String? = null
)


data class SetUsernameRequest(
    val username: String
)

data class UpdateProfileRequest(
    val profileUrl: String
)
